### PR TITLE
Clean up some duplicated code

### PR DESCRIPTION
**Pull request by jacobherrington**
### Description
Looking through this file, I noticed some identical code that could be cleaned up.

This is only better subjectively, but I didn't see anything re: refactors in the contributing doc. If Puma tends to avoid style-oriented refactors like this one without a strong rationale, feel free to close this PR.

### Your checklist for this pull request

- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [x] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: lib/puma/runner.rb [hunk 1]
```diff
@@ -15,6 +15,17 @@ def initialize(cli, events)
       @app = nil
       @control = nil
       @started_at = Time.now
+      @wakeup = nil
+    end
+
+    def wakeup!
+      return unless @wakeup
+
+      begin
```
Comment (MSP-Greg) [inline lib/puma/runner.rb:24]: Maybe remove `begin` and move the rescue to 'method level'?

File: lib/puma/runner.rb [hunk 2]
```diff
@@ -159,5 +166,13 @@ def start_server
       server.inherit_binder @launcher.binder
       server
     end
+
+    private
+
+    def ensure_output_directory_exists(path)
```
Comment (MSP-Greg) [inline lib/puma/runner.rb:172]: Maybe something like:
```ruby
def ensure_output_directory_exists(path, io_name)
```
and pass `'STDOUT'` or `'STDERR'` to it, along with changing:
```ruby
raise "Cannot redirect #{io_name} to #{path}"

**Comment by MSP-Greg** [comment]
> If Puma tends to avoid style-oriented refactors

Fine by me, not sure about others.  Most older repos always have code that can be refactored.  And, ruby/ruby is always refactoring...

**Comment by jacobherrington** [comment]
@MSP-Greg Thanks for the review, I pushed a new commit including your suggestions!

**Comment by nateberkopec** [comment]
We're 100% ok with refactors here. I wouldn't even call this style-oriented, you're removing code, not just changing quotation marks and whitespace.

**Pull request merged**